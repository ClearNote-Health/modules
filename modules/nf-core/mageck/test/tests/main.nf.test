nextflow_process {

    name "Test Process MAGECK_TEST"
    script "../main.nf"
    process "MAGECK_TEST"

    tag "modules"
    tag "modules_nfcore"
    tag "mageck"
    tag "mageck/test"

    test("mageck-test - csv") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/mus_musculus/mageck/count_table.csv', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert file(process.out.gene_summary[0][1]).readLines().first().contains("pos|p-value") },
                { assert snapshot(process.out.version).match("version") }
            )
        }

    }

    test("mageck-test-day0 - csv") {

        config "./nextflow-day0_label.config"

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/mus_musculus/mageck/count_table.csv', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert file(process.out.gene_summary[0][1]).readLines().first().contains("pos|p-value") },
                { assert snapshot(process.out.version).match("day0_version") }
            )
        }

    }

    test("mageck-test - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/mus_musculus/mageck/count_table.csv', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
