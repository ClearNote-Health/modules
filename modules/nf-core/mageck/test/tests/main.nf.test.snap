{
    "day0_version": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-21T15:29:56.274882"
    },
    "mageck-test - csv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gene_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sgrna_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.R:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,d52903c294d3f9b3ff6244878af3a757"
                ],
                "gene_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gene_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "r_script": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.R:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sgrna_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sgrna_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d52903c294d3f9b3ff6244878af3a757"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-25T09:50:40.204114"
    },
    "version": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-21T15:29:30.190372"
    }
}